src/components/app/content/cards/card/index.js

```diff
@@ -1,51 +1,60 @@
-import React, {Fragment, useEffect, useState} from 'react';
+import React, {Fragment, useCallback, useEffect, useState} from 'react';
 import './index.css'
 
 const classNames = require('classnames');
 
 function Card(props) {
     const cardData = props.cardData
-    const cardId = cardData.id
 
-    const [backupState, setBackupState] = useState({header: 'Caption', body: 'Text...'})
+    const [cardOptions, setCardOptions] = useState({checked: false, editable: false})
+    const [tempState, setTempState] = useState({header: props.cardData.header, body: props.cardData.body})
 
-    useEffect(() => {
-        props.editCard(cardId, {...cardData, ...backupState, editable: false});
-        }, [props.readonly])
+    const cancelEditing = () => setCardOptions({checked: false, editable: false})
 
     const saveCardDataChanges = () => {
-        props.editCard(cardId, {...cardData, editable: false})
-        setBackupState({header: cardData.header, body: cardData.body})
+        cancelEditing()
+        props.saveCardData(cardData.id, tempState)
     }
+    const restoreCardDataChanges = useCallback(() => {
+        cancelEditing()
+        setTempState({header: cardData.header, body: cardData.body})
+    }, [cardData.header, cardData.body])
```

Comment:
Again, in that way you'll need to add a new field reference each time new field is added to the card object. Not consistent, just set the whole tempState(better use spread operator) and check for the whole cardData changes.